### PR TITLE
Update attribute form (BOUBEST-414)

### PR DESCRIPTION
**Pull request by jorgen-norheim**
This is what has been done:

- Made some changes to models (mostly replacing null with undefined, easier to work with in react-hook-form)
- Added attribute form type and mappers between requests, form types and views
- Updated attribute form to new data model
- Added form validation and error messages
- Removed inclusive/exclusive boolean variables from value constraints (range constraints will always be inclusive)

There is still things to be done, both in design and probably also creating form components that fit the task better than the ones used now, but the form is functional, so this can be revisited later.

### CODE REVIEW

File: src/client/src/features/entities/common/form-attributes/FormAttributes.tsx [hunk 1]
```diff
@@ -1,13 +1,10 @@
 import { AttributeLibCm } from "@mimirorg/typelibrary-types";
 import { XCircle } from "@styled-icons/heroicons-outline";
 import { Flexbox, Token } from "@mimirorg/component-library";
-import { useGetAttributes } from "external/sources/attribute/attribute.queries";
-import {
-  onAddAttributes,
-  resolveSelectedAndAvailableAttributes,
-} from "features/entities/common/form-attributes/FormAttributes.helpers";
+//import { useGetAttributes } from "external/sources/attribute/attribute.queries";
```
Comment (AndyEPandy) [inline src/client/src/features/entities/common/form-attributes/FormAttributes.tsx:4]: Fjerne disse?

File: src/client/src/features/entities/common/form-attributes/FormAttributes.tsx [hunk 2]
```diff
@@ -41,25 +38,21 @@ export interface FormAttributesProps {
...
+  register, //canAddAttributes = true, //limitedAttributes = [], //canRemoveAttributes = true,
 }: FormAttributesProps) => {
   const theme = useTheme();
   const { t } = useTranslation("entities");
 
-  const attributeQuery = useGetAttributes();
-  const attributes = preprocess ? preprocess(attributeQuery.data) : attributeQuery.data ?? [];
-  const [available, selected] = resolveSelectedAndAvailableAttributes(fields, attributes);
+  //const attributeQuery = useGetAttributes();
+  //const attributes = preprocess ? preprocess(attributeQuery.data) : attributeQuery.data ?? [];
```
Comment (AndyEPandy) [inline src/client/src/features/entities/common/form-attributes/FormAttributes.tsx:49]: Samme her? Vil du ha disse med videre?

File: src/client/src/features/entities/common/form-attributes/FormAttributes.tsx (hunk 1 above)
Comment (jorgen-norheim) [reply src/client/src/features/entities/common/form-attributes/FormAttributes.tsx:4]: Dette er kode som foreløpig bare er kommentert ut for å få koden til å bygge. Tenker jeg beholder de der til vi skal endre på funksjonalitet i denne komponenten, sånn at man ser hvordan den virka før. :)

File: src/client/src/features/entities/attributes/AttributeForm.tsx [hunk 3]
```diff
@@ -53,17 +55,15 @@ export const AttributeForm = ({ defaultValues = createEmptyAttribute(), mode }:
...
+        onSubmit={handleSubmit((data) => onSubmitForm(toAttributeTypeRequest(data), mutation.mutateAsync, toast))}
       >
         {isLoading ? (
           <Loader />
         ) : (
           <Box display={"flex"} flex={2} flexDirection={"row"} gap={theme.mimirorg.spacing.multiple(6)}>
             <Flexbox flexDirection={"column"} gap={theme.mimirorg.spacing.l}>
               <Text variant={"display-small"}>{t("attribute.title")}</Text>
-              <AttributeFormBaseFields limited={mode === "edit" && query.data?.state === State.Approved} />
+              <AttributeFormBaseFields limited={false} />
```
Comment (Bouvet-MatsAlmeland) [inline src/client/src/features/entities/attributes/AttributeForm.tsx:66]: Er det meningen at denne skal være statisk?

**Pull request merged**